packages/block-editor/src/store/selectors.js

```diff
@@ -1376,6 +1376,24 @@ export function isLastBlockChangePersistent( state ) {
 	return state.blocks.isPersistentChange;
 }
 
+/**
+ * Returns the Block List settings for an array of blocks, if any exist.
+ *
+ * @param {Object}  state    Editor state.
+ * @param {Array} clientIds Block client IDs.
+ *
+ * @return {Array} Block List Settings for each of the found blocks
+ */
+export const __experimentalGetBlockListSettingsForBlocks = createSelector(
+	( state, clientIds ) => {
+		return filter( state.blockListSettings, ( value, key ) => clientIds.includes( key ) );
+	},
+	( state, clientIds ) => [
+		state.blockListSettings,
+		clientIds,
```

Comment:
I think normally we don't put selector parameters here.

Reply:
You're saying we shouldn't include `clientIds` in the array here?

I can't find any examples where this pattern is repeated so I think it must be correct. I was going on [this from the Rememo docs](https://github.com/aduth/rememo):

> ...a second function which returns the reference or array of references upon which the selector's derivation depends

I assume `clientIds` was the key factor here. Because the value should be recomputed both if the `clientIds` change or the `blockListSettings` change. 🤷‍♂

Reply:
I fixed this btw.